### PR TITLE
Hamdi/datastore

### LINKED ISSUE #168: Change structure of store/
## Summary
 
Rename `outputStore` to `dataStore` and merge in `blockStore`. Split store funcs into separate files for block, tx, fees, deposits etc.

### PR DESCRIPTION
**Pull request by hamdiallam**
Single data store for the application. `DataStore`

1. Merge `BlockStore` and `OutputStore` into a single `DataStore`. resolves  #168 
2. Merge queriers into a single querier mounted under `custom/data/...`
3. Remove `store/query.BlocksResp`. It's an extra unnecessary struct and now  `plasma.Block` includes the block height

#168

### CODE REVIEW

File: store/querier.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+	if err != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("json: %s", err)).
+			WithDefaultCodespace(DefaultCodespace)
+	}
+	return data, nil
+}
+
+func parseHeight(height string) (*big.Int, sdk.Error) {
+	h, ok := new(big.Int).SetString(height, 10)
+	if !ok || h.Sign() < 0 {
```
Comment (colin-axner) [inline store/querier.go]:
```suggestion
	if !ok || h.Sign() =< 0 {
```

File: handlers/anteHandler.go [hunk 2]
```diff
@@ -154,11 +154,11 @@ func validateConfirmSignatures(ctx sdk.Context, input plasma.Input, inputTx stor
 	return sdk.Result{}
 }
 
-func includeDepositAnteHandler(ctx sdk.Context, outputStore store.OutputStore, blockStore store.BlockStore, msg msgs.IncludeDepositMsg, client plasmaConn) (newCtx sdk.Context, res sdk.Result, abort bool) {
-	if outputStore.HasDeposit(ctx, msg.DepositNonce) {
+func includeDepositAnteHandler(ctx sdk.Context, msg msgs.IncludeDepositMsg, ds store.DataStore, client plasmaConn) (newCtx sdk.Context, res sdk.Result, abort bool) {
```
Comment (AdityaSripal) [inline handlers/anteHandler.go]: I think it makes sense to switch ds and msg arguments here

File: store/querier.go [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+}
+
+func queryTxOutput(ctx sdk.Context, ds DataStore, path []string) ([]byte, sdk.Error) {
+	if len(path) != 1 {
+		return nil, ErrInvalidPath(fmt.Sprintf("expected %s/<position>", QueryTxOutput))
+	}
+
+	pos, err := plasma.FromPositionString(path[0])
+	if err != nil {
+		return nil, ErrInvalidPath("position is encoded in the format (blocknum,txIndex,oIndex,depositNonce)")
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply store/querier.go]: on second thought, this is fine since the argument comes from the path

File: handlers/anteHandler.go (hunk 2 above)
Comment (AdityaSripal) [reply handlers/anteHandler.go]: Don't have strong opinions on this either way so long as it is consistent

File: store/querier.go [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+	}
+
+	height, err := parseHeight(path[0])
+	if err != nil {
+		return nil, err
+	}
+
+	block, ok := ds.GetBlock(ctx, height)
+	if !ok {
+		return nil, ErrDNE("plasma block %s does not exists", height)
```
Comment (colin-axner) [inline store/querier.go]:
```suggestion
		return nil, ErrDNE("plasma block %s does not exist", height)
```

**Pull request merged**